generator/testdata/google/example/library/v1/library_service.gapic.cc.baseline

```diff
@@ -9,84 +11,192 @@
 gax::StatusOr<::google::example::library::v1::Book>
 LibraryService::CreateBook(
 ::google::example::library::v1::CreateBookRequest const& request) {
-  // TODO: actual useful work, e.g. retry, backoff, metadata, pagination, etc.
+  // TODO: update metadata and idempotency, etc.
   grpc::ClientContext context;
   ::google::example::library::v1::Book response;
   grpc::Status status = stub_->CreateBook(&context, request, &response);
+
   if (status.ok()) {
     return response;
-  } else {
-    return gax::GrpcStatusToGaxStatus(status);
   }
+
+  std::string error_message;
+  // TODO: use idempotency policy to determine whether a failure is permanent
+  if (!isPermanentFailure(status)) {
+    auto retry_policy = retry_policy_->clone();
+    auto backoff_policy = backoff_policy_->clone();
+    while (retry_policy->OnFailure(status)) {
+      auto backoff_duration = backoff_policy->OnCompletion();
+      std::this_thread::sleep_for(backoff_duration);
+      status = stub_->CreateBook(&context, request, &response);
+      if (status.ok()) {
+        return response;
+      }
+    // TODO: augment error_message with result from status
+    }
+  }
+  // TODO: implement grpcStatusCodeToGaxStatusCode
+  gax::StatusCode code = grpcStatusCodeToGaxStatusCode(status.error_code());
+  return gax::Status(code, error_message);
 }
 
 gax::StatusOr<::google::example::library::v1::Book>
 LibraryService::GetBook(
 ::google::example::library::v1::GetBookRequest const& request) {
-  // TODO: actual useful work, e.g. retry, backoff, metadata, pagination, etc.
+  // TODO: update metadata and idempotency, etc.
   grpc::ClientContext context;
   ::google::example::library::v1::Book response;
   grpc::Status status = stub_->GetBook(&context, request, &response);
+
   if (status.ok()) {
     return response;
-  } else {
-    return gax::GrpcStatusToGaxStatus(status);
   }
+
+  std::string error_message;
+  // TODO: use idempotency policy to determine whether a failure is permanent
+  if (!isPermanentFailure(status)) {
+    auto retry_policy = retry_policy_->clone();
+    auto backoff_policy = backoff_policy_->clone();
+    while (retry_policy->OnFailure(status)) {
+      auto backoff_duration = backoff_policy->OnCompletion();
+      std::this_thread::sleep_for(backoff_duration);
+      status = stub_->GetBook(&context, request, &response);
+      if (status.ok()) {
+        return response;
+      }
+    // TODO: augment error_message with result from status
+    }
+  }
+  // TODO: implement grpcStatusCodeToGaxStatusCode
+  gax::StatusCode code = grpcStatusCodeToGaxStatusCode(status.error_code());
```

Comment:
Same here, I believe the Google C++ style guide requires functions and member functions to start with uppercase.

Reply:
Will change, the same caveats from `isPermanentFailure` apply here.